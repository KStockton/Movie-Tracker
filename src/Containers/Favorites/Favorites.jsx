import React, { Component } from "react";
import { connect } from "react-redux";
import { addTopMovies } from "../../Actions/index";
import CardContainer from "../../Components/CardContainer/CardContainer";

export class Favorites extends Component {

  render() {
    return (
      <section className="home-container">
        <h2 className="home-title"> </h2>
        {this.props.user.name !== undefined &&
        this.props.user.favorites.length !== undefined ? (
          <CardContainer
            topMovies={this.props.user.favorites}
            user={this.props.user}
          />
        ) : (
          <div className="no-favorites">
            Log in to see favorites!
          </div>
        )}
      </section>
    );
  }
}

const mapStateToProps = state => ({
  topMovies: state.topMovies,
  user: state.user
});

const mapDispatchToProps = dispatch => ({
  addTopMovies: movies => dispatch(addTopMovies(movies))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Favorites);
