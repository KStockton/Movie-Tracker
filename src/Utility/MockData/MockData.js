
export const uncleanMovies =
[
  {
      "vote_count": 2045,
      "id": 19404,
      "video": false,
      "vote_average": 9,
      "title": "Dilwale Dulhania Le Jayenge",
      "poster_path": "/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg",
      "original_language": "hi",
      "original_title": "दिलवाले दुल्हनिया ले जायेंगे",
      "genre_ids": [
          35,
          18,
          10749
      ],
      "backdrop_path": "/nl79FQ8xWZkhL3rDr1v2RFFR6J0.jpg",
      "adult": false,
      "overview": "Raj is a rich, carefree, happy-go-lucky second generation NRI. Simran is the daughter of Chaudhary Baldev Singh, who in spite of being an NRI is very strict about adherence to Indian values. Simran has left for India to be married to her childhood fiancé. Raj leaves for India with a mission at his hands, to claim his lady love under the noses of her whole family. Thus begins a saga.",
      "release_date": "1995-10-20"
  },
  {
      "vote_count": 12962,
      "id": 278,
      "video": false,
      "vote_average": 8.7,
      "title": "The Shawshank Redemption",
      "poster_path": "/9O7gLzmreU0nGkIB6K3BsJbzvNv.jpg",
      "original_language": "en",
      "original_title": "The Shawshank Redemption",
      "genre_ids": [
          18,
          80
      ],
      "backdrop_path": "/j9XKiZrVeViAixVRzCta7h1VU9W.jpg",
      "adult": false,
      "overview": "Framed in the 1940s for the double murder of his wife and her lover, upstanding banker Andy Dufresne begins a new life at the Shawshank prison, where he puts his accounting skills to work for an amoral warden. During his long stretch in prison, Dufresne comes to be admired by the other inmates -- including an older prisoner named Red -- for his integrity and unquenchable sense of hope.",
      "release_date": "1994-09-23"
  }]

  export const cleanMovies = 
  [{
     id: 19404,
     title: "Dilwale Dulhania Le Jayenge",
     backdrop_path: "/nl79FQ8xWZkhL3rDr1v2RFFR6J0.jpg",
     overview: "Raj is a rich, carefree, happy-go-lucky second generation NRI. Simran is the daughter of Chaudhary Baldev Singh, who in spite of being an NRI is very strict about adherence to Indian values. Simran has left for India to be married to her childhood fiancé. Raj leaves for India with a mission at his hands, to claim his lady love under the noses of her whole family. Thus begins a saga.",
     poster_path: "/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg",
     vote_average: 9,
     release_date: "1995-10-20",
     favorited: false
  },
    {
      id: 278,
      title:  "The Shawshank Redemption",
      backdrop_path: "/j9XKiZrVeViAixVRzCta7h1VU9W.jpg",
      overview: "Framed in the 1940s for the double murder of his wife and her lover, upstanding banker Andy Dufresne begins a new life at the Shawshank prison, where he puts his accounting skills to work for an amoral warden. During his long stretch in prison, Dufresne comes to be admired by the other inmates -- including an older prisoner named Red -- for his integrity and unquenchable sense of hope.",
      poster_path: "/9O7gLzmreU0nGkIB6K3BsJbzvNv.jpg",
      vote_average: 8.7,
      release_date: "1994-09-23",
      favorited: false
  }]

  export const movieProp = {
    state: {
      movie: {
        id: 2341,
        title: "HellBoy",
        backdrop_path: "/5BkSkNtfrnT",
        favorited: false,
        overview: "walk on earth",
        poster_path: "/bk8LyaMqUtaQ9hUShuvFznQYQKR.jpg",
        release_date: "2019-04-12"
      }
    }
  }
