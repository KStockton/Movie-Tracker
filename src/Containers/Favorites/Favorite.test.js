import React from 'react';
import { Favorites, mapStateToProps, mapDispatchToProps } from './Favorites';
import { shallow } from 'enzyme';
import * as MD from '../../Utility/MockData/MockData'


describe('Favorites', () => {
  let wrapper

  beforeEach(() => {
      wrapper = shallow(<Favorites user={MD.mockUser}/>)
  })

  it('Favorite component render Login in to see favorites if not recieving user props', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should return a props object with a movies array', () => {
    const mockState = {
      topMovies: [
        {'title': "SpaceJam"}
      ]
    }
    const expected = {
      topMovies: mockState.topMovies
    }
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
  })

  it('should return a p')

})
