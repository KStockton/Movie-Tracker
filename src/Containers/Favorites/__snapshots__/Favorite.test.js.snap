// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites Favorite component render Login in to see favorites if not recieving user props 1`] = `
<section
  className="home-container"
>
  <h2
    className="home-title"
  >
     
  </h2>
  <div
    className="no-favorites"
  >
    Log in to see favorites!
  </div>
</section>
`;
